rholang/src/main/scala/coop/rchain/rholang/interpreter/Registry.scala

```diff
@@ -0,0 +1,216 @@
+                      case Some(Expr(EMapBody(parMap))) =>
+                        parMap.ps.get(Expr(GByteArray(head))) match {
+                          case None => localFail()
+                          case Some(value) =>
+                            value.singleExpr() match {
+                              case Some(Expr(ETupleBody(ETuple(ps, _, _)))) =>
+                                if (ps.length != 3)
+                                  localFail()
+                                else
+                                  // The second tuple field should be a bytearray in both cases.
+                                  ps(1).singleExpr() match {
+                                    case Some(Expr(GByteArray(edgeAdditional))) =>
+                                      ps(0).singleExpr() match {
+                                        case Some(Expr(GInt(0))) =>
+                                          if (tail == edgeAdditional)
+                                            replace(data, replaceChan, dataRand).flatMap(_ =>
+                                              succeed(ret, ps(2), callRand))
+                                          else
+                                            localFail()
+                                        case Some(Expr(GInt(1))) =>
+                                          if (tail.startsWith(edgeAdditional)) {
+                                            val newKey = tail.substring(edgeAdditional.size)
+
+                                            replace(data, replaceChan, dataRand).flatMap(
+                                              _ =>
+                                                fetchData(Quote(ps(2)),
+                                                          Channel(Quote(GByteArray(newKey))),
+                                                          ret,
+                                                          callRand))
+                                          } else
+                                            localFail()
+                                        case _ => localFail()
+                                      }
+                                    case _ => localFail()
+                                  }
+                              case _ => localFail()
+                            }
+                        }
+                      case _ => localFail()
+                    }
+                  case _ => localFail()
+                }
+              case _ =>
+                localFail()
+            }
+          case _ =>
+            localFail()
+        }
+      case _ => Task.unit
+    }
+
+  val testingDispatchTable: Map[Long, Function1[RootSeq[ListChannelWithRandom], Task[Unit]]] =
+    Map(lookupRef -> lookup, lookupCallbackRef -> lookupCallback)
+}
+
+object Registry {
+  val registryRoot = GPrivate(
+    ByteString.copyFrom(
+      Base16.decode("a4fd447dedfc960485983ee817632cf36d79f45fd1796019edfb4a84a81d1697")))
```

Comment:
How was this number generated?

Reply:
kyle@clinch:~/pyrofex-devel/rchain$ echo -en "rho:io:registry" | b2sum | sed -E 's/.{64}/\0\n/'
a4fd447dedfc960485983ee817632cf36d79f45fd1796019edfb4a84a81d1697
d5fd3d8daf9f295aa590b37b50d5518803b4596ed6940aa42d46b1413a1bb16e  -

We can use something else if you want, this is a minor thing to focus on.